src/workloads/execution/ShardedGraphLookup.yml

```diff
@@ -0,0 +1,268 @@
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 10
+    Database: *Database
+    Operations:
+    # The recursive targeted queries below work as follows: for each document X in Collection0 
+    # we find the document Y in Collection0 where X's '_id' (startWith) is equal to Y's '_id'
+    # (connectToField). Then, we look at Y's 'a' (connectFromField) value and find the document Z in 
+    # Collection0 that has the same value for its '_id' (connectToField), and so on. 
+    # Because the collection is sharded by '_id' the $graphLookup is targeted towards 
+    # specific shard(s). Also, because for each document the value for 'a' is one greater than 
+    # the '_id' (and the ID fields increase monotonically), each document will have in its matches 
+    # all documents whose '_id' is greater than or equal to its own (limited by maxDepth).
+
+    # Similarly, for the recursive untargeted queries below: for each document X in Collection0
+    # we find the document Y in Collection0 where X's 'a' value (startWith) is equal to Y's 'a' 
+    # value (connectToField). Then, we look at Y's '_id' (connectFromField) and find the document Z
+    # in Collection0 that has the same value for its 'a' value (connectToField), and so on. 
+    # Because the collection is sharded by '_id' and 'a' is the connectToField, the $graphLookup
+    # is not targeted to any specific shard. Since the connectToField and connectFromField are 
+    # swapped from the targeted queries, each document will have in its matches all documents whose
+    # '_id' is less than or equal to its own (limited by maxDepth).
+
+    # For the non-recursive queries, each document will match only with itself.
+
+    # Targeted $graphLookup from sharded collection to sharded collection, no recursion
+    - OperationMetricsName: TargetedGraphLookupShardedToShardedNoRecursion
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{
+            $graphLookup: {
+              from: "Collection0", 
+              startWith: "$_id", 
+              connectFromField: "a", 
+              connectToField: "_id", 
+              as: "matches",
+              maxDepth: 0
+            }
+          }]
+        # To get meaningful results, the entire result set should fit in a single batch. This should
+        # be possible since both collections are small.
+        cursor: {batchSize: *NumDocs}
+
+    # Targeted $graphLookup from sharded collection to sharded collection, recursion allowed
+    - OperationMetricsName: TargetedGraphLookupShardedToShardedRecursionAllowed
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{
+            $graphLookup: {
+              from: "Collection0", 
+              startWith: "$_id", 
+              connectFromField: "a", 
+              connectToField: "_id", 
+              as: "matches",
+              maxDepth: 13
```

Comment:
Note to other reviewers, here was the thinking about the maxDepth value: The size of the result set will look approximately like `numDocs * sizeOfDoc * (maxDepth + 1) + numDocs * sizeOfDoc`, where the first number represents all of the matches from the right side and the second represents the documents on the left side. With the current doc size of approx 35 bytes, and a collection size of 20k, we'll get around 10MB of output data for a depth of 13.

The untargeted queries take a long time (~2 minutes). This could be caused by the collection size, which is pretty large compared to the collection sizes of recent $lookup workloads. An alternative, while still maintaining a similar result set size, could be to increase the size of each document and decrease the collection size.